src/Mado/QueryBundle/Objects/HalResponse.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+namespace Mado\QueryBundle\Objects;
+
+class HalResponse
+{
+    private $json;
+
+    private function __construct(array $json)
+    {
+        $this->json = $json;
+    }
+
+    public static function fromJson(string $json)
+    {
+        return self::fromArray(json_decode($json, true));
```

Comment:
Can we add an exception or something to detect if is an array and a test for it?

Reply:
There is type hinting for this.